File: desktop/src/main/java/com/techfinance/pessoal/desktop/service/UserApiService.java
package com.techfinance.pessoal.desktop.service;

import com.google.inject.Inject;
import com.google.inject.Singleton;
import com.techfinance.pessoal.desktop.dto.response.UserResponse;
import com.techfinance.pessoal.desktop.infrastructure.http.ApiClient;

@Singleton
public class UserApiService {

    private final ApiClient apiClient;

    @Inject
    public UserApiService(ApiClient apiClient) {
        this.apiClient = apiClient;
    }

    public UserResponse findByUsername(String username) {
        return apiClient.get("/user/" + username, UserResponse.class);
    }
}

File: web/src/services/userApiService.ts
import { apiClient } from '../infrastructure/http/apiClient'
import type { UserResponse } from '../types/user'

class UserApiService {
  findByUsername(username: string): Promise<UserResponse> {
    return apiClient.get<UserResponse>(`/user/${username}`)
  }
}

export const userApiService = new UserApiService()
